### ISSUE TITLE
Advertising not really stopped?

### ISSUE THREAD

**Issue by randalln** [weliem/blessed-kotlin#14]
I'm refactoring my Android app to use your sweet library, but I encountered an issue with toggling Bluetooth off and on that I was able to reproduce with the peripheral example app.  I'm new to BLE, so I'll offer what I was able gather at least.

**Steps to reproduce**

1. Start peripheral app
2. Turn Bluetooth off on the device
3. Background the app
4. Turn Bluetooth on
5. Foreground the app

**Logs**

2024-04-20 12:50:43.659 11009-11023 BluetoothGattServer     com.welie.btserver                   D  onServiceAdded() - handle=143 uuid=00001805-0000-1000-8000-00805f9b34fb status=0
2024-04-20 12:50:44.147 11009-11009 AdvertiseSettings       com.welie.btserver                   D  setTxPowerLevel: 2
2024-04-20 12:50:44.150 11009-11009 BluetoothAdapter        com.welie.btserver                   D  isLeEnabled(): ON
2024-04-20 12:50:44.156 11009-11009 BluetoothLeAdvertiser   com.welie.btserver                   D  TxPower == ADVERTISE_TX_POWER_MEDIUM
2024-04-20 12:50:44.157 11009-11009 BluetoothAdapter        com.welie.btserver                   D  isLeEnabled(): ON
2024-04-20 12:50:44.176 11009-11009 BluetoothP...ralManager com.welie.btserver                   I  advertising started
2024-04-20 12:50:47.008 11009-11085 ProfileInstaller        com.welie.btserver                   D  Installing profile for com.welie.btserver
2024-04-20 12:51:13.812 11009-11009 BluetoothP...ralManager com.welie.btserver                   D  bluetooth turning off
2024-04-20 12:51:13.890 11009-11009 BluetoothP...ralManager com.welie.btserver                   D  bluetooth turned off
2024-04-20 12:51:13.890 11009-11009 BluetoothP...ralManager com.welie.btserver                   I  advertising stopped
2024-04-20 12:52:21.869 11009-11009 VRI[MainActivity]       com.welie.btserver                   D  visibilityChanged oldVisibility=true newVisibility=false
2024-04-20 12:52:21.905 11009-11009 VRI[MainActivity]       com.welie.btserver                   D  Not drawing due to not visible
2024-04-20 12:52:21.935 11009-11009 WindowOnBackDispatcher  com.welie.btserver                   W  sendCancelIfRunning: isInProgress=falsecallback=android.view.ViewRootImpl$$ExternalSyntheticLambda11@aff331
2024-04-20 12:52:29.077 11009-11009 BluetoothP...ralManager com.welie.btserver                   D  bluetooth turning on
2024-04-20 12:52:29.079 11009-11009 BluetoothP...ralManager com.welie.btserver                   D  bluetooth turned on
2024-04-20 12:52:39.372 11009-11009 AdvertiseSettings       com.welie.btserver                   D  setTxPowerLevel: 2
2024-04-20 12:52:39.374 11009-11009 BluetoothAdapter        com.welie.btserver                   D  isLeEnabled(): ON
2024-04-20 12:52:39.378 11009-11009 BluetoothP...ralManager com.welie.btserver                   E  advertising failed with error 'ALREADY_STARTED'

**Comment by randalln** [comment]
Things don't explode when I add  `bluetoothLeAdvertiser.stopAdvertising(advertiseCallback)` to BluetoothPeripheralManager#cancelAllConnectionsWhenBluetoothOff, but is that the right thing to do? 🤔

**Comment by weliem** [comment]
Hi, thanks for raising this issue. What happens when BLE is switched off depends a bit on the phone brand. Nonetheless, I just pushed [ac5c0ec](https://github.com/weliem/blessed-kotlin/commit/ac5c0ecf65e9927f61e976b2ab3021e313589266) that improves the behavior. Like your PR, the advertising is explicitly stopped now and I also added a callback for adapter state changes.

Things should be more stable now.
